tests/system-fetcher/tests/handler_test.go

```diff
@@ -658,6 +744,8 @@ func TestSystemFetcherDuplicateSystemsForTwoTenants(t *testing.T) {
 	require.NotEmpty(t, template2.ID)
 
 	triggerSync(t, tenant.TestTenants.GetDefaultTenantID())
+	waitForApplicationsToBeProcessed(ctx, t, tenant.TestTenants.GetDefaultTenantID(), 2)
+	triggerSync(t, tenant.TestTenants.GetSystemFetcherTenantID())
```

Comment:
Do we need to wait for this sync as well?

Reply:
Yes, at it may proceed longer.